### PR TITLE
Add EIP: Ground Zero

### PR DESCRIPTION
**Pull request by ProphetZX**
EIP-7488 (Ground Zero) suggests important governance changes, as required by the community.

### CODE REVIEW

**Comment by abcoathup** [comment]
Yet another spam PR for EIP-0 (Ground Zero), similar to #7278 #7332

File: EIPS/eip-0.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+eip: 0
```
Comment (SamWilsn) [inline EIPS/eip-0.md:2]:
```suggestion
eip: 7488
```

File: EIPS/eip-0.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Simple Summary
+
+This EIP "Ground Zero" suggests to switch mainnet to permissioned to lower gas prices and transaction costs. The adjustment requires protcol and governance changes. 
+
+## Abstract
+
+The EIP idea is to brake the IPFS and Swarm store and broadcasting and instead suggests scalling down ETHEREUM ressources, so that the community may be served in a more sustainable way. In addition, to mitigate the currently existing problem with increasingly high gas prices, ETHEREUM would benefit from switching to a permissioned mainnet (after successfully deploying a testnet version). This would greatly lower the number of requests and decrease the load for operating nodes. A further improvement would be to remove legacy APIs, so nodes no longer answer unnecessary public API requests. 
+
+## Rationale
```
Comment (SamWilsn) [inline EIPS/eip-0.md:18]: You'll need to follow the header order from EIP-1.

File: EIPS/eip-0.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the [DAO hack and resulting fork](https://www.coindesk.com/learn/understanding-the-dao-attack/), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things.
```
Comment (SamWilsn) [inline EIPS/eip-0.md:20]: Please expand abbreviations ("p.p.") the first time they are used.

Comment (SamWilsn) [inline EIPS/eip-0.md:20]:
```suggestion
Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the DAO hack and resulting fork (see [EIP-779](./eip-779.md)), illegal contracts and privacy intrusion, data theft, artificial and malicious content, and many more such things. 
```

File: EIPS/eip-0.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Rationale
+
+Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the [DAO hack and resulting fork](https://www.coindesk.com/learn/understanding-the-dao-attack/), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things. 
+
+Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the ETHEREUM ecosystem and infrastructure of important ressources and instead developing use cases for the original community would be a remedy for almost all of these problems.
```
Comment (SamWilsn) [inline EIPS/eip-0.md:22]: Please change "ETHEREUM" to "Ethereum" in the entire document.

```suggestion
Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the Ethereum ecosystem and infrastructure of important resources and instead developing use cases for the original community would be a remedy for almost all of these problems.
```

File: EIPS/eip-0.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Mainnet (ID: 1) MUST be switched to permissioned. In part, this may be achieved by halting IPFS, Swarm, and Whisper, after a successfull testnet version has been deployed. This will affect existing DApps, but almost none of these have been used in the past by the community anyway. So the impact of this protocol change should be minor. For active transactions and smart contracts, opcode 0x00 SHOULD be executed. As a necessary and sufficient condition, opcode 0xff for the zero address SHALL be exectued as well.
```
Comment (SamWilsn) [inline EIPS/eip-0.md:28]: > Mainnet (ID: 1) MUST be switched to permissioned.

This does not provide enough detail. What does a permissioned Ethereum look like? Are you talking about [clique](https://eips.ethereum.org/EIPS/eip-225)? Something else? To be an EIP, a proposal needs to be specified to a sufficient level of detail that it can be implemented.

> This will affect existing DApps, but almost none of these have been used in the past by the community anyway.

This belongs in the backwards compatibility or rationale sections.

> For active transactions and smart contracts, opcode 0x00 SHOULD be executed. As a necessary and sufficient condition, opcode 0xff for the zero address SHALL be exectued as well.

Could you clarify this a bit more in the document? Do you mean that instead of the currently deployed bytecode, these opcodes will be executed instead?

File: EIPS/eip-0.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Potentially, an adjustment to devp2p, EIP-8, would be required as well and Oracle connections should be switched.
```
Comment (SamWilsn) [inline EIPS/eip-0.md:30]: You'll need to describe the exact changes you want made.

File: config/.codespell-whitelist
Comment (SamWilsn) [inline config/.codespell-whitelist:1]: Don't change this file.

File: config/eip-editors.yml
Comment (SamWilsn) [inline config/eip-editors.yml:1]: Don't change this file.

**Review by SamWilsn — CHANGES_REQUESTED** [review]

**Comment by Pandapip1** [comment]
This user's GitHub profile seems to indicate that they are attempting to sneak in a change to the EIP Editors file.

@ProphetZX if this is not the case, please open a new PR that only touches files inside the `EIPS` (or `assets` directory, if needed). Please re-read EIP-1.

**Review by Pandapip1 — COMMENTED** [review]
Content-wise, I would recommend looking at EIP-225. It could significantly shorten your EIP if you reference it.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please fix the walidator errors

**Comment by ProphetZX** [comment]
> Please fix the walidator errors

All checks have successfully passed.

**Comment by Pandapip1** [comment]
> I couldn’t find an EIP-225, do you mean EIPS/eip-2255.md? If so, be aware that the wallet key injections, permissions… without the need to access sensitive methods is not the same as what EIP Ground Zero suggests.

No, I meant [EIP-225](https://eips.ethereum.org/EIPS/eip-225)

**Comment by ProphetZX** [comment]
> > I couldn’t find an EIP-225, do you mean EIPS/eip-2255.md? If so, be aware that the wallet key injections, permissions… without the need to access sensitive methods is not the same as what EIP Ground Zero suggests.
> 
> No, I meant [EIP-225](https://eips.ethereum.org/EIPS/eip-225)

I found it, cheers for the link! In that regard, Goerli is still active, right? I‘m underthe impression that activity in the Goerli testnet somewhat ebbed away over the last months.

**Comment by Pandapip1** [comment]
Goerli has been deprecated, and so users have correctly migrated away from it. However, the ERC-223 protocol itself is still implemented in most clients.

**Comment by ProphetZX** [comment]
> Goerli has been deprecated, and so users have correctly migrated away from it. However, the ERC-223 protocol itself is still implemented in most clients.

Ok, thanks for the clarifications. Would you mind reviewing the EIP?

File: EIPS/eip-7488.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+Mainnet (ID: 1) MUST be switched to permissioned. The permissioned ethereum blockchain then either applies proof-of-authority consensus (see [EIP-225](./eip-225.md)), or alternatively, a new consensus mechanism. In part, this may be achieved by halting IPFS, Swarm, and Whisper, after a successfull testnet version has been deployed, for which Goerli or Sepolia may provide blue prints to some degree.
+
+Before halting IPFS and Swarm, packages SHOULD be transfered via p2p to the original owner, including the original data.
+
+Since for some existing dApps, token holdings would likely be exposed to such a protocol change, it is REQUIRED to decide what happens to affected [ERC-721](./eip-721.md) tokens and [ERC-20](./eip-20.md) coins. A potential solution would be to gather all tokens and decide step-by-step which ones would still be valid, could be transfered and/or sold to benefit the community, and which ones should be burned.
+
+## Rationale
+
+Alternatively, if this EIP "Ground Zero" should fail and no similar solution will be found, a temporary halt of the ethereum blockchain may be considered instead.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:35]: This is not what the Rationale section is for. The Rationale section is for discussing design decisions. In particular--why is PoA the best solution to the problems described in the Motivation? Why does IPFS, a completely unrelated technology, have to be "stopped"? Why should the new validators go through all the assets and decide whose to discard?

All of these questions should be answered for me to consider this Rationale section complete.

File: EIPS/eip-7488.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the ethereum ecosystem and infrastructure of important ressources and instead developing use cases for the original community would be a remedy for almost all of these problems.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Mainnet (ID: 1) MUST be switched to permissioned. The permissioned ethereum blockchain then either applies proof-of-authority consensus (see [EIP-225](./eip-225.md)), or alternatively, a new consensus mechanism. In part, this may be achieved by halting IPFS, Swarm, and Whisper, after a successfull testnet version has been deployed, for which Goerli or Sepolia may provide blue prints to some degree.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:27]: A few comments here:

- You need to specify when the transition will take effect. It has to be deterministic and computable.
- You're using too many words. Just say something to the effect that that Ethereum will transition to PoA using EIP-225 as its consensus mechanism.
- You need to specify the initial EIP-225 validator set.
- IPFS and co. are unrelated to Ethereum, and due to their distributed nature, it's not possible to "stop" them. That section has to be removed.
- Goerli never used ERC-225

File: EIPS/eip-7488.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+Before halting IPFS and Swarm, packages SHOULD be transfered via p2p to the original owner, including the original data.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:29]:
```suggestion
```
IPFS is an unrelated technology. If you want to halt IPFS, [bring it up with them](https://github.com/ipfs/specs).

File: EIPS/eip-7488.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+Since for some existing dApps, token holdings would likely be exposed to such a protocol change, it is REQUIRED to decide what happens to affected [ERC-721](./eip-721.md) tokens and [ERC-20](./eip-20.md) coins. A potential solution would be to gather all tokens and decide step-by-step which ones would still be valid, could be transfered and/or sold to benefit the community, and which ones should be burned.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:31]: What about ERC-1155 tokens? What about other tokens?

How do you enumerate all the assets? Specifically, how to you get a list of every single ERC-20 or ERC-721 contract?

What happens when you do? Who decides? Do the validators all vote on it? If so, is there a time limit? What percentage have to vote? What format do the votes take?

Why do the assets need to be transferred at all? Will they not still work?

File: EIPS/eip-7488.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+## Motivation
+
+Since its humble beginnings, the ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such an open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the DAO hack and resulting fork (see [EIP-779](./eip-779.md)), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:19]: - There's a typo. Ethereum should be capitalized.
- What are the consensus problems? You never specify.
- I can't understand what you mean by "false ownership claims of assets." Could you rephrase that a little better?
- The DAO hack happened in 2016.
- Who defines what contracts are illegal?
- Are there any specific cases of privacy intrusions that you know about? The only thing I can think of is that the blockchain is completely public, and transitioning to a permissioned system doesn't change that.
- You already said privacy intrusions. Data theft is wholly covered at this point.
- What is "artificial content?"

File: EIPS/eip-7488.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+## Motivation
+
+Since its humble beginnings, the ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such an open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the DAO hack and resulting fork (see [EIP-779](./eip-779.md)), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things. 
+
+Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the ethereum ecosystem and infrastructure of important ressources and instead developing use cases for the original community would be a remedy for almost all of these problems.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:21]:
```suggestion
```
As previously mentionned, this will have to go.

Also, it's false. IPFS has, if anything, lessened the burden on the Ethereum blockchain by offloading some of the data storage.

File: EIPS/eip-7488.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
+---
+eip: 7488
+title: Ground Zero
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:3]: This is a non-descriptive title. Please use something like
```suggestion
title: Upgrade consensus to Proof-of-Authority
```

File: EIPS/eip-7488.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+description: Implements protocol and governance changes required by the community
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:4]: This is a non-descriptive description. Please use something like
```suggestion
description: Specifies changes needed to transition Ethereum from Proof-of-Stake to Proof-of-Authority
```

File: EIPS/eip-7488.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+discussions-to: https://ethereum-magicians.org/t/eip-ground-zero/15480
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-06-30
+---
+
+## Abstract
+
+This EIP "Ground Zero" suggests to switch mainnet to permissioned to lower gas prices, transaction costs, and improve privacy. The adjustment requires protcol and governance changes. The EIP idea is braking and stopping the IPFS and Swarm store and broadcasting and instead suggests scalling down ethereum ressources, so that the community may be served in a more **sustainable way** in the future. In addition, to mitigate the currently existing problem with increasingly high gas prices, ethereum would benefit from switching to a permissioned mainnet (after successfully deploying a testnet version). This would greatly lower the number of requests and decrease the load for operating nodes. A further improvement would be to remove legacy APIs, so nodes no longer answer unnecessary public API requests.
```
Comment (Pandapip1) [inline EIPS/eip-7488.md:15]: - You should not reference your own EIP by your title. Just use "This EIP", without the title.
- This EIP doesn't suggest. Use "This EIP switches mainnet..."
- "The adjustment requires protcol and governance changes." is implied by this being `category: Core`. Remove it.
- Remove all mentions of IPFS/Swarm.
- "Braking" is ill-defined here. Use a different word, like "stopping."
- There's a typo in scaling.
- Actually, what you are describing doesn't really scale Ethereum up or down. Remove that part entirely.
- You mention sustainability here, but I don't see that in the Motivation. It's an argument for this EIP. Put that in the Motivation.
- Ethereum is capitalized.
- Testnets are an unrelated, irrelevant implementation detail. Remove that mention of them.
- Remove the last sentence mentionning "removing legacy APIs". It's unrelated to this EIP.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
